cmd/mimir/config-descriptor.json

```diff
@@ -7389,6 +7389,17 @@
           "fieldFlag": "query-frontend.parallelize-shardable-queries",
           "fieldType": "boolean"
         },
+        {
+          "kind": "field",
+          "name": "rewrite_propagate_matchers",
```

Comment:
I don't think we need to include "rewrite" as part of the setting name here, _how_ the matchers are propagated doesn't really matter - just that they are.

Reply:
We plan to add a 2nd kind of optimization pass in this rewrite middleware after merging this PR, so I thought it'd be clearer to specify that both of them are part of the rewrite middleware but focus on different areas. If you still feel that both of them should omit the "rewrite_" part I'm open to changing this.